### PR TITLE
refactoring for separate controllers

### PR DESCRIPTION
**Pull request by mallett002**
This is a work in progress for creating the `getUserExerciseLog` controller

### CODE REVIEW

**Comment by mallett002** [comment]
This is a work in progress for creating the `getUserExerciseLog` controller

**Review by mmallett — APPROVED** [review]

File: controllers/getUserExerciseLog.js [hunk 1]
```diff
@@ -1,6 +1,83 @@
...
+                error: err || `Something went wrong finding the user with id ${userId}`
+            });
+        });
+};
+
+function keepGreaterThanOrEqual(log) {
+    let date = log.date;
+    
+    // remove "th" from "Wednesday, Feb, 6th 2019" bc new Date(dateString) doesn't work with 'th' or 'fth' in it
+    const dateEndingRegex = /\d+(rd|th|st|nd)/; // capture the date number ending (eg. 'th' from '6th')
```
Comment (mmallett) [inline controllers/getUserExerciseLog.js]: Does moment support this?

File: controllers/getUserExerciseLog.js [hunk 2]
```diff
@@ -1,6 +1,83 @@
...
+        })
+        .catch(err => {
+            res.status(500).send({
+                error: err || `Something went wrong finding the user with id ${userId}`
+            });
+        });
+};
+
+function keepGreaterThanOrEqual(log) {
+    let date = log.date;
```
Comment (mmallett) [inline controllers/getUserExerciseLog.js]: Why is your date not stored in a Date object or a format easily converted into a Date object?

File: controllers/getUserExerciseLog.js [hunk 3]
```diff
@@ -1,6 +1,83 @@
...
+                // if numberItems > limit:
+                    // pop off end while numberItems !== limit
+                // else just return items
+            }
+
+            if (!logs.length) {
+                return res.json({
+                    message: `User with id ${userId} has no logs with given parameters`
+                });
+            }
```
Comment (mmallett) [inline controllers/getUserExerciseLog.js:48]: More concise:
```
let filterdLogs = user.logs
  .filter(log => !fromDate || date >= fromDate) // assuming both are Date
  .filter(log => !toDate || date <= toDate);

if (limit) {
  filteredLogs = filteredLogs.slice(0, 20);
}

if (!logs.length) {
  // no results response
} 
```

File: models/user.model.js [hunk 4]
```diff
@@ -4,7 +4,8 @@ const Schema = mongoose.Schema;
 const exerciseSchema = new Schema({
     description: { type: String, required: true },
     duration: { type: Number, required: true },
-    date: { type: String, required: true }
+    date: { type: String, required: true },
+    dateInfo: { type: Number, required: true }
```
Comment (mallett002) [inline models/user.model.js:8]: Calling `moment()` returns a number.

**Pull request merged**